Docs rework help

TC72 opened the pull request:
Changes for the developer documentation, mainly helping out with grammar but also trying to simplify the technical side where possible.

TC72 committed:
Fixed some grammar and tried to simplify the technical wording

Conni2461 commented on developers.md:
```diff
@@ -151,41 +154,45 @@ pair in our table that we pass into `pickers.new`, for example after `sorter`.
+`move_selection_{next,previous}`. So in most cases you'll want to return `true`.
+If the function does not return anything then an error is thrown. 
+
+The `attach_mappings` function has two parameters, `prompt_bufnr` is the buffer number
+of the prompt buffer, which we can use to get the pickers object and `map` is a table
+we can use to map actions or functions to arbitrary key sequences.
```
`map` is a function that we can use to map functions / actions to keys. Its not a table.

TC72 committed:
Fixed the mistake of calling `map` a table instead of a function

TC72 commented:
I've fixed calling `map` a table instead of a function.
Also fixed the sentence "that you already have some knowledge", was missing "you".

Could you also check how I explain what's passed to `display` if you use a function?
I think I understand it correctly but am I right in saying `display` won't be passed in the table to the `display` function as it would be a reference to itself?

```
The `display` key is required and is either a string or a `function(tbl)`,
where `tbl` is the table returned by `entry_maker`. So in this example `tbl` would
give our `display` function access to `value` and `ordinal`. (Not `display` as 
that would be a reference to itself.)
```

Conni2461 commented:
I think thats good. The last sentence is wrong tho. 

> (Not `display` as that would be a reference to itself.)

Because you also have access to display itself. But you shouldn't call that function. Or maybe you can if you want to implement display recursive :rofl:

Conni2461 commented:
So lets remove that sentence. Otherwise its good to go imo.

We can always iterate over it later :laughing: 

Also hacktoberfest is coming and people love to fix typos ;)

TC72 committed:
Removed line about passing display to display

TC72 commented:
OK, that's done.
I've asked TJ of he can find the drawing I mentioned.

I'll also look at adding screenshots of what should be seen as you run this code.

Conni2461 commented:
Thanks :)

Conni2461 commented:
But lets do that some other time. Otherwise we never merge it :laughing:

TC72 commented:
Oh yes, we'll do that later.
